Removing some NS prefixes from folders, files and documentation

hartbit opened the pull request.

hartbit commented:
@ianpartridge It touches so many files that it may be wise to merge it soon before another commit creates any conflits 😄

hartbit commented:
@swift-ci please smoke test

parkera commented:
Already has a conflict. =)

ianpartridge commented:
@hartbit I agree we should merge it ASAP. If you can fix it up, I'll do my best...

ikesyo commented:
Splitting up into smaller chunks will avoid such conflicts. :wink:

ikesyo commented on Foundation/ByteCountFormatter.swift:
```diff
@@ -91,7 +91,7 @@ open class ByteCountFormatter : Formatter {
     private let numberFormatter = NumberFormatter()
     
-    /* Shortcut for converting a byte count into a string without creating an NSByteCountFormatter and an NSNumber. If you need to specify options other than countStyle, create an instance of NSByteCountFormatter first.
+    /* Shortcut for converting a byte count into a string without creating an ByteCountFormatter and an Number. If you need to specify options other than countStyle, create an instance of ByteCountFormatter first.
```
This should still be `NSNumber`, right?

hartbit replied:
Yep, fixed.

ikesyo commented on Foundation/NSURLSession/URLSessionTask.swift:
```diff
@@ -521,7 +521,7 @@ open class URLSessionStreamTask : URLSessionTask {
 
 /* Key in the userInfo dictionary of an NSError received during a failed download. */
-public let URLSessionDownloadTaskResumeData: String = "NSURLSessionDownloadTaskResumeData"
+public let URLSessionDownloadTaskResumeData: String = "URLSessionDownloadTaskResumeData"
```
This should be kept as is for compatibility with Darwin. Also related to [SR-3403](https://bugs.swift.org/browse/SR-3403).

hartbit replied:
Yes, I was careful to avoid those modifications, but one slipped through the cracks. Thx!

ikesyo commented on TestFoundation/TestNotificationQueue.swift:
```diff
@@ -117,7 +117,7 @@ FIXME SR-4280 timeouts in TestNSNotificationQueue tests
             queue.enqueue(notification, postingStyle: .now, coalesceMask: [], forModes: [runLoopMode])
             queue.enqueue(notification, postingStyle: .now)
-            // here we post notification for the NSRunLoopCommonModes. It shouldn't have any affect, because the timer is scheduled in NSDefaultRunLoopMode.
+            // here we post notification for the NSRunLoop.commonModes. It shouldn't have any affect, because the timer is scheduled in NSDefaultRunLoopMode.
```
> NSRunLoop.commonModes

This should be `RunLoopMode.commonModes`.

https://developer.apple.com/documentation/foundation/runloopmode/1408609-commonmodes

ikesyo replied:
> NSDefaultRunLoopMode

Same as above.

ikesyo commented on TestFoundation/TestNotificationQueue.swift:
```diff
@@ -117,7 +117,7 @@ FIXME SR-4280 timeouts in TestNSNotificationQueue tests
             // post 2 notifications for the NSDefaultRunLoopMode mode
```
> NSDefaultRunLoopMode

This should be `RunLoopMode.defaultRunLoopMode`.

https://developer.apple.com/documentation/foundation/runloopmode/1409732-defaultrunloopmode

hartbit commented:
@ikesyo I fixed the errors you spotted. Thx!
@ianpartridge @parkera I fixed the conflicts.

Further questions:
* Can a CI build be triggered to test the PR like we have for swiftpm?
* I have merge rights, so I can handle that once it has been accepted. Is that okay with you?

ianpartridge commented:
@swift-ci please test

ianpartridge commented:
@swift-ci please test

ianpartridge commented:
@hartbit 

```
ninja: warning: multiple rules generate ../Ninja-ReleaseAssert/foundation-linux-x86_64/Foundation/Foundation/URL.swift.o. builds involving this target will not be correct; continuing anyway
ninja: error: 'Foundation/Cache.swift', needed by '../Ninja-ReleaseAssert/foundation-linux-x86_64/Foundation/Foundation/Cache.swift.o', missing and no known rule to make it
/home/buildnode/jenkins/workspace/swift-corelibs-foundation-PR-Linux/swift/utils/build-script: fatal error: command terminated with a non-zero exit status 1, aborting
```

hartbit commented:
@swift-ci please test

hartbit commented:
@swift-ci please test

hartbit commented:
@swift-ci please test

hartbit commented:
Is there a way to execute `build.py` locally on macOS against the current snapshot?

hartbit commented:
@swift-ci please test

ianpartridge commented:
@swift-ci please test and merge

ianpartridge commented:
@swift-ci please test and merge

ianpartridge commented:
@swift-ci please test and merge

ianpartridge commented:
@swift-ci please test and merge

ianpartridge commented:
I've merged this before it gets stale again.  It's hard to review something so large but I did eyeball the whole PR.  If there are any issues we can fix those up individually.

Thanks @hartbit
